pkg/pushsync/pushsync.go

```diff
@@ -183,78 +186,55 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 
 	price := ps.pricer.Price(chunkAddress)
 
-	// if the peer is closer to the chunk, AND it's a full node, we were selected for replication. Return early.
-	if p.FullNode {
-		if closer, _ := p.Address.Closer(chunkAddress, ps.address); closer {
-
-			ps.metrics.HandlerReplication.Inc()
+	if ch.Replicate {
 
-			var err error
-			defer func() {
-				if err != nil {
-					ps.metrics.HandlerReplicationErrors.Inc()
-				}
-			}()
+		ps.metrics.HandlerReplication.Inc()
```

Comment:
Before actually storing this we should also check if this chunk is actually within our storage radius. This could be used by some malicious node or some node with incorrect state to falsely make a peer store a chunk.